src/tools/r2rdump/R2RDump.cs

```diff
@@ -459,12 +428,11 @@ private int Run(string[] args)
             // open output stream
             if (_outputFilename != null)
             {
-                FileStream fileStream = new FileStream(_outputFilename, FileMode.Create, FileAccess.Write);
-                _writer = new R2RWriter(fileStream);
+                _writer = File.CreateText(_outputFilename);
             }
             else
             {
-                _writer = new R2RWriter(Console.OpenStandardOutput());
+                _writer = Console.Out;
             }
 
             foreach (string filename in _inputFilenames)
```

Comment:
Could you wrap the foreach loop in a `try {` block and then the _writer.Close() in a `finally {` block?